extensions/amp-copy/1.0/component.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as Preact from '#preact';
+import {
+  copyTextToClipboard,
+  isCopyingToClipboardSupported,
+} from '#core/window/clipboard';
+import {
+  useCallback,
+  useImperativeHandle,
+  useLayoutEffect,
+  useState,
+} from '#preact';
+import {forwardRef} from '#preact/compat';
+import {useStyles} from './component.jss';
+import objstr from 'obj-str';
+
+/**
+ * @param {!CopyDef.Props} props
+ * @param {{current: ?CopyDef.CopyApi}} ref
+ * @return {PreactDef.Renderable}
+ */
+export function CopyWithRef({children, sourceId, text, ...rest}, ref) {
```

Comment:
nit: it may be worth aliasing this to `textProp` so that you can then use the variable name `text` in smaller scope locations instead of `staticText` etc.